### PR TITLE
Covid dashboard

### PR DESCRIPTION
**Pull request by davidmiller**

### CODE REVIEW

File: plugins/covid/management/commands/calculate_covid_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        CovidDashboard.objects.all().delete()
+        print('Deleted Dashboard objects')
+
+        tested = Patient.objects.filter(
+            lab_tests__test_name=CORONAVIRUS_TEST_NAME).distinct()
+        print('Found Tested Patients')
```
Comment (fredkingham) [inline plugins/covid/management/commands/calculate_covid_dashboard.py]: Not actually true, 
"Queried for tested patients" is more accurate

File: plugins/covid/management/commands/calculate_covid_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        print('Found Positive Patients')
+
+        negative = tested.filter(
+            lab_tests__observation__observation_value__in=NEGATIVE_TEST_VALUES
+        ).exclude(id__in=positive_ids)
+
+        CovidDashboard.objects.create(
+            patients_tested=tested.count(),
+            positive=positive.count(),
+            negative=negative.count()
```
Comment (fredkingham) [inline plugins/covid/management/commands/calculate_covid_dashboard.py:37]: We expect
patients_tested == positive + negative?

(is it worth sending an email if they don't?)

File: plugins/covid/management/commands/calculate_covid_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        tested = Patient.objects.filter(
+            lab_tests__test_name=CORONAVIRUS_TEST_NAME).distinct()
+        print('Found Tested Patients')
+
+        positive = tested.filter(
+            lab_tests__observation__observation_value__in=POSITIVE_TEST_VALUES
+        )
+        positive_ids = [p.id for p in positive]
+        print('Found Positive Patients')
```
Comment (fredkingham) [inline plugins/covid/management/commands/calculate_covid_dashboard.py]: Does the distinct() clause definitely carry over to here? I think it _should_ but I'm not 100%

IE are we sure that if a patient has multiple positive test values they won't be duplicated here?

File: plugins/covid/models.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from django.db import models
+
+
+class CovidDashboard(models.Model):
+    """
+    Stores figures for the Covid 19 Dashboard in the
+    database when they take a significant amount of time
+    to calculate.
+    """
+    patients_tested = models.IntegerField()
```
Comment (fredkingham) [inline plugins/covid/models.py:13]: I think we should add auto add fields (ie created timestamps)

File: plugins/covid/plugin.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+        dashboard = menus.MenuItem(
+            href='/#/covid-19/',
+            display='Covid 19',
+            icon='fa fa-dashboard',
+            activepattern='/#/covid-19/'
+        )
+
+        profile = UserProfile.objects.get(user=user)
+        if profile.roles.filter(name=COVID_ROLE).exists():
```
Comment (fredkingham) [inline plugins/covid/plugin.py:27]: if I'm `user.profile.roles.filter(name=COVID_ROLE).exist()` looks a bit nicer but I that's just my aesthetic opinion (very unimportant)

File: plugins/covid/management/commands/calculate_covid_dashboard.py (hunk 3 above)
Comment (davidmiller) [reply plugins/covid/management/commands/calculate_covid_dashboard.py]: yes

File: plugins/covid/models.py (hunk 4 above)
Comment (davidmiller) [reply plugins/covid/models.py:13]: Not sure there's any point. Created is always arbitrarily 12 minutes past the last hour b/c the cron job and the delete all. We only need one row in this table, we're not storing history.

**Pull request merged**